### PR TITLE
[serving] Adds promtheus metrics support

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Comment by zachgk** [comment]
Is there a reason we can't add the prometheus support as a plugin?

**Comment by frankfliu** [comment]
We can add as a plugin, but the `log4j2.xml` reference to `prometheus` is problematic if the plugin is not there.
The reason add prometheus as a different project is purely because complier flag issue. log4j annotation processor will generate compiler warning and cause error with "-Werr" flag.

File: prometheus/src/main/java/ai/djl/prometheus/MetricExporter.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Writes prometheus metrics to {@code OutputStream}.
+     *
+     * @param os the {@code OutputStream} to write
+     * @param set the filter names
+     * @throws IOException throws if failed to write
+     */
+    public static void export(OutputStream os, Set<String> set) throws IOException {
+        if (!Boolean.parseBoolean(Utils.getEnvOrSystemProperty("SERVING_PROMETHEUS"))) {
+            os.write(
```
Comment (zachgk) [inline prometheus/src/main/java/ai/djl/prometheus/MetricExporter.java:42]: Can we make this return an exception and therefore have the HTTP API return a failure error code in this case?

Comment (frankfliu) [reply prometheus/src/main/java/ai/djl/prometheus/MetricExporter.java]: Fixed

**Review by zachgk — APPROVED** [review]

**Pull request merged**